README.md

```diff
@@ -31,14 +69,393 @@ Permissions to allow the user to interact with the support center:
+  aws:
+    regions:
+    - name: us-east-1
+  awsCredentialSecret:
+    name: aws
+    namespace: {NameSpace cluster is being built in}
+  legalEntity:
+    id: 00000000000000
+    name: {Legal Entity Name}
+status:
+  conditions:
+  - lastProbeTime: 2019-07-16T13:52:02Z
+    lastTransitionTime: 2019-07-16T13:52:02Z
+    message: Attempting to claim account
+    reason: AccountClaimed
+    status: "True"
+    type: Unclaimed
+  - lastProbeTime: 2019-07-16T13:52:03Z
+    lastTransitionTime: 2019-07-16T13:52:03Z
+    message: Account claimed by osd-creds-mgmt-fhq2d2
+    reason: AccountClaimed
+    status: "True"
+    type: Claimed
+  state: Ready
+```
+# 3. The controllers
+
+## 3.1. AccountPool Controller
+
+The accountpool-controller is triggered by an accountpool CR or an account CR. It is responsible for filling the Acccount Pool by generating new account CRs. 
+
+It looks at the accountpool CR *spec.poolSize* and it ensures that the number of unclaimed accounts matchs the number of the poolsize. If the number of unclaimed accounts is less then the poolsize it creates a new account CR for the account-controller to process.
+
+### 3.1.1. Constants and Globals
+```
+emailID = "osd-creds-mgmt"
+```
+
+### 3.1.2. Status
+
+Updates accountPool cr 
+```
+  claimedAccounts: 4
+  poolSize: 3
+  unclaimedAccounts: 3
+```
+
+*claimedAccounts* are any accounts with the `status.Claimed=true`
+*unclaimedAccounts* are any accounts with `status.Claimed=false` and `status.State!="Failed"`
+*poolSize* is the poolsize from the accountPool spec
+
+### 3.1.3. Metrics
+
+Updated in the accountPool-controller
+
+```
+MetricTotalAccountCRs
+MetricTotalAccountCRsUnclaimed
+MetricTotalAccountCRsClaimed
+MetricTotalAccountPendingVerification
+MetricTotalAccountCRsFailed
+MetricTotalAccountCRsReady
+MetricTotalAccountClaimCRs
+
+```
+
+## 3.2. Account Controller
+
+The account-controller is triggered by an account CR. It is responsible for following behaviours:
+
+If the *awsLimit* set in the constants is not exceeded
+1. Creates a new account in the organization belonging to credentials in secret `aws-account-operator-credentials` 
+2. Configure Users from *iamUserNameUHC* and *iamUserNameSRE*
+    * Creates IAM user in new account 
+    * Attaches Admin policy
+    * Generates a secret access key for the user 
+    * Stores user secret in a AWS secret
+3. Creates STS CLI Credentials for SRE
+4. Creates and Destroys EC2 instances
+5. Creates aws support case to increase account limits
+
+### 3.2.1. Additional Functionailty 
+
+* If `status.RotateCredentials == true` the account-controller will refresh the STS Cli Credentials
+* If the account's `status.State == "Creating"` and the account is older then the *createPendTime* constant the account will be put into a `failed` state
+* If the account's `status.State == AccountReady && spec.ClaimLink != ""` it sets `status.Claimed = true`
+
+### 3.2.2. Constants and Globals
```

Comment:
Can we add descriptions for all of these?

Reply:
I think we should but in the next refactor. I'd rather get this merged and put tickets in to update sections that may need updating or need additional information